lib/bit-interlude.js

```diff
@@ -38,25 +37,35 @@ module.exports = class BitInterlude {
         continue
       }
 
-      return this.drop === false
+      return r.value
     }
 
     return this.bitfield.get(index)
   }
 
   setRange (start, end, value) {
-    assert(this.drop !== value || this.ranges.length === 0)
-    assert(value === true || this.ranges.length === 0)
-
-    this.drop = value === false
+    if (start === end) return
 
     let r = null
 
     for (let i = 0; i < this.ranges.length; i++) {
       r = this.ranges[i]
 
       // if already inside, stop
-      if (r.start <= start && end <= r.end) return
+      if (r.start <= start && end <= r.end) {
+        if (value === r.value) return
+
+        const lower = { start: r.start, end: start, value: r.value }
+        const upper = { start: end, end: r.end, value: r.value }
+
+        let remove = 1
+
+        if (lower.start < lower.end) this.ranges.splice(i++, remove--, lower)
+        this.ranges.splice(i++, remove, { start, end, value })
```

Comment:
the ++ is sus to me, we use i again below but normally you'd i-- it as you are removing from the array

Reply:
only one of them removes from the array, but they all add to it, so it ends up being: 

```
  arr = [0]
  arr.splice(i++, 1, 1)
  arr.splice(i++, 0, 2)
  arr.splice(i, 0, 3)

  // arr === [1, 2, 3]
```